handlers/auth.go

```diff
@@ -0,0 +1,82 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/AlonSerrano/GolangBootcamp/models"
+	"github.com/AlonSerrano/GolangBootcamp/service"
+	"github.com/AlonSerrano/GolangBootcamp/usecase"
+
+	"github.com/labstack/echo/v4"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type authHandler struct {
+	client *mongo.Collection
+}
+
+//NewAuthHandler NewAuthHandler init
+func NewAuthHandler() *authHandler {
+	return &authHandler{client: service.UseUserTable()}
+}
+
+// SignIn handler that invokes signIn function
+// @Summary SignIn a user into the app
+// @Description By sending email and password it will return a token object
+// @Tags user
+// @Param userParams body models.SignInReq true "User params"
+// @Produce json
+// @Success 200 {object} models.TokenInfoResp
+// @Router /user/signIn [post]
+func (ah *authHandler) SignIn(c echo.Context) error {
+	jsonMap := make(map[string]string)
+	err := json.NewDecoder(c.Request().Body).Decode(&jsonMap)
+	if err != nil {
+		return err
+	} else {
```

Comment:
We don't need an `else` here since we're returning in the last statement of the `if`.